data_browser/src/ExperimentPage.js

```diff
@@ -184,18 +433,80 @@ function renderExperimentStatus(events) {
 
   const duration = (endTime.getTime() - startTime.getTime()) / 1000;
 
-  const lastStatus = <span className={"status-" + lastEvent.status}>{lastEvent.status}</span>;
-  const lastMessage = lastEvent.message;
-  return (<div className="experiment-step-status">
-    Status: {lastStatus}{lastMessage ? ": " + lastMessage : ""}&nbsp;
-    {renderDate(lastEvent.date)} &mdash; {renderDuration(duration)}
-  </div>);
+  const statusIcon = statusIcons[lastEvent.status] || lastEvent.status;
+
+  const lastStatus = <span className={"status-" + lastEvent.status}>{statusIcon}</span>;
+  const temporalInfo = `last updated ${renderDate(lastEvent.date)}, lifetime is ${renderDuration(duration)}`;
+  return <span className="status-container">
+    <a href={viewStatusUrl(step)} target="_blank" title={`View raw JSON status of this step ${temporalInfo}`}>
+      {lastStatus}
+    </a>
+  </span>;
 }
 
-function renderValue(value) {
-  const str = JSON.stringify(value);
-  if (str.startsWith("\"") && str.endsWith("\"")) {
-    return str.slice(1, -1);
+function pathJoin(path, file) {
+  return path + (path.endsWith("/") ? "" : "/") + file;
+}
+
+function apiStatusUrl(step) {
+  const statusPath = pathJoin(step.output_path, ".executor_status");
+  return apiViewUrl({path: statusPath, count: 100});
+}
+
+function apiResultsUrl(step) {
+  const resultsPath = pathJoin(step.output_path, "results.json");
+  return apiViewUrl({path: resultsPath, count: 100});
+}
+
+function viewStatusUrl(step) {
+  const statusPath = pathJoin(step.output_path, ".executor_status");
+  return viewSingleUrl(statusPath);
+}
+          
+function viewInfoUrl(step) {
+  const infoPath = pathJoin(step.output_path, ".executor_info");
+  return viewSingleUrl(infoPath);
+}
+          
+function viewOutputPathUrl(step) {
+  return viewSingleUrl(step.output_path);
+}
+
+function extractRayRelativePath(path) {
+  // Given caller path, extract the relative path
+  // Input: /tmp/ray/session_2024-10-17_20-58-29_674266_488/runtime_resources/working_dir_files/_ray_pkg_38803023dcc3288a/experiments/scratch.py
+  // Output: experiments/scratch.py
+  return path.replace(/.*_ray_pkg_\w+\//, "");
+}
+
+function renderStagedValue(valueOrValues) {
+  // valueOrValues could be a number or [{start, value}, {start, value}, ...]
+  // e.g., for batch size
+  if (typeof valueOrValues === "number") {
+    return valueOrValues;
+  }
+  if (Array.isArray(valueOrValues)) {
+    const content = joinSpans(valueOrValues.map((item) => <span><sub>{item.start}</sub>{item.value}</span>), "→");
+    return <span>[{content}]</span>;
   }
-  return str;
+  return "???";
+}
+
+function computeNumTokens({numSteps, batchSize, seqLen}) {
```

Comment:
I hope I didn't screw this up

Reply:
i believe this is correct (and had o4 check too)